as/plugins/org.jboss.ide.eclipse.as.wtp.core/src/org/jboss/ide/eclipse/as/core/server/UnitedServerListenerManager.java

```diff
@@ -46,7 +46,9 @@ public static UnitedServerListenerManager getDefault() {
 	}

```

Comment:
This file is a perpetual disaster for me. Discovered situations where the UnitedServerListenerManager is added twice as a listener to server objects, so this needed to be cleaned up and synchronized a bit.